pkg/cloud/packet/client.go

```diff
@@ -144,23 +147,29 @@ func (p *Client) NewDevice(ctx context.Context, req CreateDeviceRequest) (*packn
 		facility = req.MachineScope.PacketMachine.Spec.Facility
 	}
 
-	serverCreateOpts := &packngo.DeviceCreateRequest{
-		Hostname:      req.MachineScope.Name(),
-		ProjectID:     req.MachineScope.PacketCluster.Spec.ProjectID,
-		Facility:      []string{facility},
-		BillingCycle:  req.MachineScope.PacketMachine.Spec.BillingCycle,
-		Plan:          req.MachineScope.PacketMachine.Spec.MachineType,
-		OS:            req.MachineScope.PacketMachine.Spec.OS,
-		IPXEScriptURL: req.MachineScope.PacketMachine.Spec.IPXEUrl,
-		Tags:          tags,
-		UserData:      userData,
+	hostname := req.MachineScope.Name()
+	serverCreateOpts := metal.CreateDeviceRequest{
+		DeviceCreateInFacilityInput: &metal.DeviceCreateInFacilityInput{
+			Hostname: &hostname,
+			Facility: metal.FacilityInputFacility{
+				String: &facility,
+			},
+			BillingCycle:    &req.MachineScope.PacketMachine.Spec.BillingCycle,
+			Plan:            req.MachineScope.PacketMachine.Spec.MachineType,
+			OperatingSystem: req.MachineScope.PacketMachine.Spec.OS,
+			IpxeScriptUrl:   &req.MachineScope.PacketMachine.Spec.IPXEUrl,
```

Comment:
I think all of these pointers will be safe, but it's worth calling out that in the conversion from packngo to metal-go we may have differences in which properties are references and which have the `omitempty` struct field tag. 

For example, if `POST ...devices {"ipxe_script_url": "", ...}` were problematic I think we may have more work to do here.

https://github.com/equinix-labs/metal-go/blob/0061d54b9022e0d74f3f47243bff98c77c42a686/metal/v1/model_device_create_input.go#L41 -- we would omit a `nil` pointer, but not a pointer to an empty string.

I believe the EM API will ignore an empty string value in this case, so I think we'll be ok here.

Reply:
The local end-to-end tests pass, so these particular pointer conversions seem to be OK.